nodedb.go

```diff
@@ -671,6 +698,41 @@ func (ndb *nodeDB) decrVersionReaders(version int64) {
 	}
 }
 
+func (ndb *nodeDB) flushVersion(version int64, rootHash []byte) error {
+	// Create new recentDB and snapshotDB batch objects. We don't use the current
+	// batch objects of the nodeDB as we don't want to write state prematurely or
+	// have conflicting state.
+	//
+	// NOTE: We ignore any write that happen to recentBatch.
+	rb := ndb.recentDB.NewBatch()
+	sb := ndb.snapshotDB.NewBatch()
+
+	// save branch, the root, and the necessary orphans
+	node := ndb.GetNode(rootHash)
+	ndb.savebranchBatch(node, true, rb, sb)
+
+	traverseOrphansVersionFromDB(ndb.recentDB, version, func(k, v []byte) {
```

Comment:
@AdityaSripal is this the correct logic for flushing orphans? Does the test case cover this case? Are there corner-cases we aren't considering?

Reply:
I'm not very familiar with the orphan logic. Is the purpose of `flushVersion` to take a version committed to memory and save it to disk instead? If so, why are we only considering orphans, shouldn't we consider all live objects?

Reply:
The purpose is to flush a previously _committed_ version that is in memory, yes. The flushing logic here is identical to what you'll find in `SaveVersion` so that's what I went off of. It's also what @AdityaSripal suggested to do.